ecosystem/typescript/sdk/src/providers/indexer.ts

```diff
@@ -0,0 +1,82 @@
+import { AnyNumber } from "../bcs/types";
+import axios from "axios";
+import { print } from "graphql";
+
+import { MaybeHexString } from "../hex_string";
+import { GetAccountCurrentTokens, GetTokenActivities } from "../indexer/generated/queries";
+import { GetAccountCurrentTokensQuery, GetTokenActivitiesQuery } from "../indexer/generated/operations";
+
+interface PaginationArgs {
+  offset?: AnyNumber;
+  limit?: number;
+}
+
+type GraphqlQuery = {
+  query: string;
+  variables?: {};
+};
+/**
+ * Provides methods for retrieving data from Aptos Indexer.
+ * For more detailed Queries specification see {@link https://cloud.hasura.io/public/graphiql?endpoint=https://indexer.mainnet.aptoslabs.com/v1/graphql}
+ */
+export class IndexerClient {
+  endpoint: string;
+
+  /**
+   * @param endpoint URL of the Aptos Indexer API endpoint.
+   */
+  constructor(endpoint: string) {
+    this.endpoint = endpoint;
+  }
+
+  /**
+   * Builds a axios client call to fetch data from Aptos Indexer.
+   *
+   * @param graphqlQuery A GraphQL query to pass in the `data` axios call.
+   */
+  private async queryIndexer(graphqlQuery: GraphqlQuery): Promise<any> {
+    try {
+      const { data } = await axios({
+        url: this.endpoint,
+        method: "post",
+        data: graphqlQuery,
+      });
+      if (data.errors) {
+        console.log("data error", data.errors);
+        return data.errors;
+      }
+      return data.data;
+    } catch (error) {
+      console.log("error", error);
```

Comment:
you're swallowing the error here and returning implicitly undefined. I think you should propagate the error to avoid silent failures.